### ISSUE TITLE
BPFD integration with NetObserv

### ISSUE THREAD

**Issue by astoycos** [bpfd-dev/bpfd#322]
This Issue will track putting together a POC of integrating the [neobserv](https://github.com/netobserv) Operator with BPFD. 

Specifically it will swap out using cilium/ebpf and the netlink library to load and attach it's TC bpf programs and instead rely on BPFD and it's associated kubernetes objects.

**Comment by astoycos** [comment]
Working through this while also refactoring our helper API in #325

**Comment by astoycos** [comment]
The netobserv operator can now deploy via bpfd with the helpers defined in #325 that work can be tracked here -> 
https://github.com/astoycos/network-observability-operator/tree/bpfd-integration (keep an out out for a PR soon)

Specifically with #325 and [this](https://github.com/astoycos/network-observability-operator/tree/bpfd-integration) branch a user can do the following on an existing bpfd KIND cluster

```
Netobserv Demo

** Build images 

`IMG="quay.io/bpfd/network-observability-operator:test" make image-build image-push`

** For existing KIND cluster

1. Make deploy-all

`make deploy-all` 

2. Make deploy-kind

`IMG="quay.io/bpfd/network-observability-operator:test" make deploy-kind`

3. Sub correct AGENT image

`kubectl -n netobserv set env deployment/netobserv-controller-manager -c "manager" RELATED_IMAGE_EBPF_AGENT="quay.io/bpfd/netobserv-ebpf-agent:main"` 

4. Deploy samples 

`make deploy-sample-cr`
```


Now I just need the necessary changes for the `netobserv-ebpf-agent` to properly interface with the maps maintained via bpfd, I will update here as I go.

**Comment by astoycos** [comment]
Working through the ebpf agent code now, we're slightly blocked on https://github.com/redhat-et/bpfd/issues/338 but for a minimal POC we should be alright.

**Comment by astoycos** [comment]
Ok so I've pretty much got this done with within https://github.com/netobserv/netobserv-ebpf-agent/pull/101 and https://github.com/netobserv/network-observability-operator/pull/288 but the POC does have some stipulations :) 

- #325  Is needed in BPFD 
- https://github.com/aya-rs/aya/pull/520 Is needed in Aya
- We can't configure the constants yet this will be fixed by https://github.com/redhat-et/bpfd/issues/338
```
	constSampling      = "sampling"
	constTraceMessages = "trace_messages"
``` 
 https://github.com/netobserv/netobserv-ebpf-agent/blob/main/pkg/ebpf/tracer.go#L68
- Currently we need to manually specify the interfaces where we wan't this to be loaded this will be fixed in https://github.com/redhat-et/bpfd/issues/324 i.e 
```
...
spec:
  namespace: netobserv
  deploymentModel: DIRECT
  agent:
    type: EBPF
    ebpf:
      imagePullPolicy: Always
      sampling: 50
      cacheActiveTimeout: 5s
      cacheMaxFlows: 100000
      interfaces: [ "eth0", "veth5871bad5" ]
...
```

Otherwise This integration should leave the existing functionality completely alone in the case that bpfd isn't deployed :) 

<img width="1366" alt="image" src="https://user-images.githubusercontent.com/23533892/223785422-d3ed73e5-5ccb-4f1b-b72a-2a2b8fea2f5e.png">

```
[astoycos@localhost bpfd-operator]$ kubectl get bpfprogramconfig
NAME                                          AGE
netobserv-flow-monitor-egress-eth0            21h
netobserv-flow-monitor-egress-veth5871bad5    21h
netobserv-flow-monitor-ingress-eth0           21h
netobserv-flow-monitor-ingress-veth5871bad5   21h

[astoycos@localhost bpfd-operator]$ kubectl get bpfProgram
NAME                                                             AGE
netobserv-flow-monitor-egress-eth0-kind-control-plane            21h
netobserv-flow-monitor-egress-veth5871bad5-kind-control-plane    21h
netobserv-flow-monitor-ingress-eth0-kind-control-plane           21h
netobserv-flow-monitor-ingress-veth5871bad5-kind-control-plane   21h
```

**Comment by astoycos** [comment]
Initial POC was fleshed out above, For any future work I will open up a new issue :)
